TRLevelControl/Control/TR1LevelControl.cs

```diff
@@ -360,9 +360,84 @@ protected override void Read(BinaryReader reader)
         }
     }
 
-    protected override void Write(BinaryWriter writer)
+    protected override void Write(TRLevelWriter writer)
     {
-        writer.Write(_level.Serialize());
+        writer.Write(_level.NumImages);
+        foreach (TRTexImage8 tex in _level.Images8) { writer.Write(tex.Serialize()); }
+
+        writer.Write(_level.Unused);
+
+        writer.Write(_level.NumRooms);
+        foreach (TRRoom room in _level.Rooms) { writer.Write(room.Serialize()); }
```

Comment:
I'm not a fan of putting multiple statements in a single line; how about converting the internal statement to an expression with the help of LINQ's ForEach?

```c#
_level.Rooms.ForEach(room => writer.Write(room.Serialize());
```

(Possibly needs a `.ToList()` call or an extension method on `IEnumerable`.)

Reply:
These statements will be getting eliminated soon :) This line will become something like `writer.Write(_level.Rooms)` and that will go off and determine what's needed, sharing out the logic between the different room types.